src/starlite_saqlalchemy/service/generic.py

```diff
@@ -64,6 +75,17 @@ async def list(self, **kwargs: Any) -> list[T]:
         """
         return []
 
+    async def list_and_count(self, **kwargs: Any) -> tuple[Sequence[T], int]:
+        """
+        Args:
+            **kwargs: Keyword arguments for filtering.
+
+        Returns:
+            List of instances and count of total collection, ignoring pagination.
+        """
+        collection, count = await asyncio.gather(self.list(**kwargs), self.count(**kwargs))
```

Comment:
I'd originally done something like this, but it looks like there were changing that impact this pattern.

See this note here: https://docs.sqlalchemy.org/en/20/_modules/examples/asyncio/gather_orm_statements.html

Reply:
Good pickup, this isn't meant to be called by `service.sqlalchemy.RepositoryService`, and should be overidden to call `repository.list_and_count()` which would bypass this.

Having said that though, this is probably a bit too cute even for the base service type, seeing that the methods it call don't do anything..